### PR TITLE
metricsbp: Add Timer.OverrideStartTime/ObserveWithEndTime

### PR DESCRIPTION
**Pull request by fishy**

### CODE REVIEW

**Review by pacejackson — CHANGES_REQUESTED** [review]
Can you also update https://github.com/reddit/baseplate.go/blob/master/metricsbp/baseplate_hooks.go to use a Timer plus OverrideStartTime rather than a histogram?

**Comment by fishy** [comment]
> Can you also update https://github.com/reddit/baseplate.go/blob/master/metricsbp/baseplate_hooks.go to use a Timer plus OverrideStartTime rather than a histogram?

Done. Also added `ObserveWithEndTime`.

File: metricsbp/baseplate_hooks.go [hunk 1]
```diff
@@ -62,9 +60,9 @@ func (h *spanHook) OnCreateChild(parent, child *tracing.Span) error {
 // OnPostStart starts the timer.
 func (h *spanHook) OnPostStart(span *tracing.Span) error {
 	if span.StartTime().IsZero() {
-		h.start = time.Now()
+		h.timer.OverrideStartTime(time.Now())
```
Comment (pacejackson) [inline metricsbp/baseplate_hooks.go:63]:
```suggestion
		h.timer.Start()
```

**Review by pacejackson — APPROVED** [review]
One small suggestion, lgtm otherwise.

**Review by konradreiche — APPROVED** [review]

**Pull request merged**